src/implementation/MassaWalletProviders.ts

```diff
@@ -0,0 +1,106 @@
+import { uid } from 'uid';
+import { ICustomEventMessageResponse } from '../interfaces/ICustomEventMessageResponse';
+import { ICustomEventMessageRequest } from '../interfaces/ICustomEventMessageRequest';
+import { AvailableCommands } from './Commands';
+import { IRegisterEvent } from '../interfaces';
+
+const MASSA_WINDOW_OBJECT_PRAEFIX = 'massaWalletProvider';
+
+type CallbackFunctionVariadicAnyReturn = (
+  error: Error | null,
+  result: any,
+) => any;
+
+// =========================================================
+
+export class MassaWalletProviders {
+  private static instance: MassaWalletProviders;
+  private registeredProviders: { [key: string]: string } = {};
+  private pendingRequests: Map<string, CallbackFunctionVariadicAnyReturn>;
+
+  public static init(): MassaWalletProviders {
+    if (!MassaWalletProviders.instance) {
+      MassaWalletProviders.instance = new MassaWalletProviders();
+    }
+
+    return MassaWalletProviders.instance;
+  }
+
+  private constructor() {
+    this.getWalletProviders = this.getWalletProviders.bind(this);
+    this.handleResponseFromContentScript =
+      this.handleResponseFromContentScript.bind(this);
+    this.sendMessageToContentScript =
+      this.sendMessageToContentScript.bind(this);
+
+    this.pendingRequests = new Map<string, CallbackFunctionVariadicAnyReturn>();
+
+    // global event target to use for all wallet provider
+    window.massaWalletProvider = new EventTarget();
+
+    // hook up register handler
+    (window.massaWalletProvider as EventTarget).addEventListener(
+      'register',
+      (evt: CustomEvent) => {
+        const payload: IRegisterEvent = evt.detail;
+        const extensionEventTarget = new EventTarget();
+        window[`${MASSA_WINDOW_OBJECT_PRAEFIX}-${payload.eventTarget}`] =
+          extensionEventTarget;
+        this.registeredProviders[payload.providerName] = payload.eventTarget;
+      },
+    );
+
+    // start listening to messages from content script
+    (window.massaWalletProvider as EventTarget).addEventListener(
+      'message',
+      this.handleResponseFromContentScript,
+    );
+  }
+
+  // send a message from the webpage script to the content script
+  public sendMessageToContentScript(
+    providerName: string,
+    command: AvailableCommands,
+    params: object,
+    responseCallback: CallbackFunctionVariadicAnyReturn,
+  ) {
+    const requestId = uid();
+    const eventMessageRequest: ICustomEventMessageRequest = {
+      params,
+      requestId,
+    };
+    this.pendingRequests.set(requestId, responseCallback);
+
+    if (!Object.values(AvailableCommands).includes(command)) {
+      throw new Error(`Unknown command ${command}`);
+    }
+
+    // dispatch an event to the window specific provider object
+    const isDispatched = (
```

Comment:
this variable is defined but not used, why the linter doesn't spotted this? same line 103 `const deleted`

Reply:
its being used now